lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -399,16 +355,24 @@ extendVerificationKeyPool ix pool
 -- >>> role @chain
 -- ...
 role :: forall (c :: Role). Typeable c => Role
-role =
-    case typeRep (Proxy :: Proxy c) of
-        t | t == typeRep (Proxy :: Proxy 'UtxoInternal) ->
-            UtxoInternal
-          | t == typeRep (Proxy :: Proxy 'UtxoExternal) ->
-            UtxoExternal
-          | t == typeRep (Proxy :: Proxy 'MultisigScript) ->
-            MultisigScript
-        _ ->
-            MutableAccount
+role =  fromJust (tryUtxoExternal <|> tryUtxoInternal <|> tryMultisigScript <|> tryMutableAccount)
```

Comment:
`fromMaybe (error "role: unmatched type" <> show typeRep (Proxy :: Proxy c))`, for clearer error message if this goes wrong?

(Or maybe `showsTypeRep` is the right function)

Reply:
done in https://github.com/input-output-hk/cardano-wallet/pull/2508/commits/37e818cc20a5248c85f8645e712295541941e614